config/log.go

```diff
@@ -0,0 +1,144 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "go.opentelemetry.io/contrib/config"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/noop"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func loggerProvider(cfg configOptions, res *resource.Resource) (log.LoggerProvider, shutdownFunc, error) {
+	if cfg.opentelemetryConfig.LoggerProvider == nil {
+		return noop.NewLoggerProvider(), noopShutdown, nil
+	}
+	opts := []sdklog.LoggerProviderOption{
+		sdklog.WithResource(res),
+	}
+	var errs []error
+	for _, processor := range cfg.opentelemetryConfig.LoggerProvider.Processors {
+		sp, err := logProcessor(cfg.ctx, processor)
+		if err == nil {
+			opts = append(opts, sdklog.WithProcessor(sp))
+		} else {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return noop.NewLoggerProvider(), noopShutdown, errors.Join(errs...)
+	}
+
+	mp := sdklog.NewLoggerProvider(opts...)
+	return mp, mp.Shutdown, nil
+}
+
+func logProcessor(ctx context.Context, processor LogRecordProcessor) (sdklog.Processor, error) {
+	if processor.Batch != nil && processor.Simple != nil {
+		return nil, errors.New("must not specify multiple log processor type")
+	}
+	if processor.Batch != nil {
+		exp, err := logExporter(ctx, processor.Batch.Exporter)
+		if err != nil {
+			return nil, err
+		}
+		return batchLogProcessor(processor.Batch, exp)
+	}
+	if processor.Simple != nil {
+		exp, err := logExporter(ctx, processor.Simple.Exporter)
+		if err != nil {
+			return nil, err
+		}
+		return sdklog.NewSimpleProcessor(exp), nil
+	}
+	return nil, fmt.Errorf("unsupported log processor type %v", processor)
```

Comment:
```suggestion
	return nil, fmt.Errorf("unsupported log processor type %[1]#v (%[1]T)", processor)
```

Reply:
did you mean to leave both v and T in here? with both it ends up logging something like:

```
"unsupported log processor type {<nil> <nil> <nil>} (config.LogRecordProcessor)"
```

Reply:
Yeah, I was mostly thinking we need to include the type to support the rest of the message. If including the instance value as well isn't going to be helpful I'm fine removing that.

Reply:
i updated the error message to specify the type of processors that are expected instead. the type of processor isn't support helpful and neither are the null values